source/dotnet/Library/AdaptiveCards/StringSizeWithUnitConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace AdaptiveCards
+{
+    public class StringSizeWithUnitConverter : JsonConverter, ILogWarnings
+    {
+        public List<AdaptiveWarning> Warnings { get; set; } = new List<AdaptiveWarning>();
+
+        readonly JsonSerializer defaultSerializer = new JsonSerializer();
+
+        public override bool CanConvert(Type objectType)
+        {
+            // Only use this converter for string types that match our format
+            return (objectType == typeof(string));
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (reader.TokenType == JsonToken.String)
+            {
+                var dimension = defaultSerializer.Deserialize(reader) as string;
+                if(uint.TryParse(dimension.Substring(0, dimension.Length - 2), out uint dimensionInPix)) {
+                    return dimensionInPix;
```

Comment:
NIT: bracket formatting

Reply:
thank you

---
In reply to: [213857621](https://github.com/Microsoft/AdaptiveCards/pull/1840#discussion_r213857621) [](ancestors = 213857621)

Reply:
i rebased it; you have to instantiate new codeflow session to see the diff
[https://github.com/Microsoft/AdaptiveCards/pull/1840](https://github.com/Microsoft/AdaptiveCards/pull/1840)


---
In reply to: [213860361](https://github.com/Microsoft/AdaptiveCards/pull/1840#discussion_r213860361) [](ancestors = 213860361,213857621)